Sources/SwiftSyntaxMacroExpansion/MacroSystem.swift

```diff
@@ -320,10 +320,25 @@ private func expandExtensionMacro(
   indentationWidth: Trivia
 ) throws -> CodeBlockItemListSyntax? {
   let extendedType: TypeSyntax
-  if let identified = attachedTo.asProtocol(NamedDeclSyntax.self) {
-    extendedType = "\(identified.name.trimmed)"
-  } else if let ext = attachedTo.as(ExtensionDeclSyntax.self) {
+  if let ext = attachedTo.as(ExtensionDeclSyntax.self) {
     extendedType = "\(ext.extendedType.trimmed)"
+  } else if let identified = attachedTo.asProtocol(NamedDeclSyntax.self) {
+    var types: [TokenSyntax] = []
+    types.append(identified.name.trimmed)
+
+    var possibleParent = attachedTo.parent
+    while let parent = possibleParent {
+      possibleParent = parent.parent
+      if let parent = parent.asProtocol(NamedDeclSyntax.self) {
+        types.append(parent.name.trimmed)
+      }
+    }
+
+    var baseType: TypeSyntax = TypeSyntax(IdentifierTypeSyntax(name: types.popLast()!))
+    for type in types.reversed() {
```

Comment:
I feel this is more straight forward.
```suggestion
    while let type = types.popLast() {
```

Reply:
Sure 👍